File: AdvancedScada.Controls/AlarmMan/Designers/ListViewDesignerForm.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using AdvancedScada.Controls.AlarmMan.Designers;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Deployment;
using System.Diagnostics;
using System.Drawing;

using System.IO;
using System.Linq;
using System.Windows.Forms;
using System.Xml;
using System.Xml.Linq;

namespace AdvancedHMIControls
{
    public partial class ListViewDesignerForm
    {
        public ListViewDesignerForm()
        {
            InitializeComponent();
        }

        #region Properties
        private iniClass inicls = new iniClass();
        private AlarmSummay m_ControlToEdit;
        public AlarmSummay ControlToEdit
        {
            get
            {
                return m_ControlToEdit;
            }
            set
            {
                m_ControlToEdit = value;
            }
        }

        private void BtnOK_Click(object sender, EventArgs e)
        {
            try
            {
                m_ControlToEdit._nListViewColumns.Clear();
                for (int i = 0; i < ListBoxSelected.Items.Count; i++)
                {
                    m_ControlToEdit._nListViewColumns.Add(Convert.ToString(ListBoxSelected.Items[i]));
                }
                // Create a file to write to.
                if (File.Exists(path) == false)
                {
                    File.CreateText(path);
                    string ListBoxSelectedItemsCount = ListBoxSelected.Items.Count.ToString();
                    inicls.SetIniValue("Display Format", " ListBoxSelectedItemsCount", ListBoxSelectedItemsCount, path);
                    for (int i = 0; i < ListBoxSelected.Items.Count; i++)
                    {
                        inicls.SetIniValue("Display Format", "ListBoxSelected.Items" + Convert.ToString(i), m_ControlToEdit._nListViewColumns[i], path);
                    }
                }
                else
                {
                    string ListBoxSelectedItemsCount = ListBoxSelected.Items.Count.ToString();
                    inicls.SetIniValue("Display Format", " ListBoxSelectedItemsCount", ListBoxSelectedItemsCount, path);
                    for (int i = 0; i < ListBoxSelected.Items.Count; i++)
                    {
                        inicls.SetIniValue("Display Format", "ListBoxSelected.Items" + Convert.ToString(i), m_ControlToEdit._nListViewColumns[i], path);
                    }
                    inicls.SetIniValue("Alarm Status", ChkAlarmOff.Name, $"{ ChkAlarmOff.Checked}", path);
                    inicls.SetIniValue("Alarm Status", ChkAlarmOn.Name, $"{ChkAlarmOn.Checked}", path);
                    inicls.SetIniValue("Alarm Status", ChkAlarmVariation.Name, $"{ ChkAlarmVariation.Checked}", path);
                    inicls.SetIniValue("Alarm Status", ChkAlarmAck.Name, $"{ChkAlarmAck.Checked}", path);

                }
                m_ControlToEdit.RListViewColumns();
                Close();
            }
            catch (Exception ex)
            {
                return;
            }
        }

        private void BtnCancel_Click(object sender, EventArgs e)
        {
            Close();
        }
        private ListBox Lst = new ListBox();

        private void Button2_Click(object sender, EventArgs e)
        {
            try
            {
                string Selected_txt = Convert.ToString(ListBoxSelected.SelectedItem);

                if (Selected_txt != null) // التحقق من عدم خلو اختيار قيمة فارغة
                {

                    if (!this.ListBoxNonSelected.Items.Contains(Selected_txt)) // يتم التحقق هل النص موجود سلفا باللست بوكس أما لا حتى لا يضاف مرة أخرى
                    {

                        this.ListBoxNonSelected.Items.Add(Selected_txt);
                        ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex);
                    }

                }

            }
            catch (Exception ex)
            {
                return;
            }


        }

        private void Button3_Click(object sender, EventArgs e)
        {
            try
            {

                int MyIndex = 0;
                string MyItem = null;
                MyIndex = ListBoxSelected.SelectedIndex;
                MyItem = ListBoxSelected.Text;

                if (MyIndex == 0)
                {
                    return;
                }

                ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex);
                ListBoxSelected.Items.Insert(MyIndex - 1, MyItem);
                ListBoxSelected.SelectedIndex = MyIndex - 1;
            }
            catch (Exception ex)
            {
                return;
            }



        }

        private void ListBoxSelected_SelectedIndexChanged(object sender, EventArgs e)
        {
            //If (ListBoxNonSelected.Items.Contains(ListBoxSelected.SelectedItem)) Then
            //    '' The item Is already exist in the listbox collection 
            //    Exit Sub
            //Else
            //    '' The item Is Not exist in the listbox collection 
            //    ListBoxNonSelected.Items.Add(ListBoxSelected.SelectedItem)
            //End If
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            try
            {
                int MyIndex = 0;
                string MyItem = null;
                MyIndex = ListBoxSelected.SelectedIndex;
                MyItem = ListBoxSelected.Text;

                if (MyIndex == ListBoxSelected.Items.Count - 1)
                {
                    return;
                }

                ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex);
                ListBoxSelected.Items.Insert(MyIndex + 1, MyItem);
                ListBoxSelected.SelectedIndex = MyIndex + 1;
            }
            catch (Exception ex)
            {
                return;
            }

        }
        private string path = "C:\\Settings.ini";
        private List<string> nListBoxSelected = new List<string>();
        private void ListViewDesignerForm_Load(object sender, EventArgs e)
        {
            try
            {


                for (int i = 0; i < ListBoxSelected.Items.Count; i++)
                {
                    nListBoxSelected.Add(Convert.ToString(ListBoxSelected.Items[i]));
                }
                m_ControlToEdit._nListViewColumnsColor.Clear();
                ListBoxSelected.Items.Clear();
                // Open the file to read from.

                if (File.Exists(path) == true)
                {
                    var ItemsCount = inicls.GetIniValue("Display Format", "ListBoxSelectedItemsCount", path);
                    for (int i = 0; string.CompareOrdinal(i.ToString(), ItemsCount) < 0; i++)
                    {
                        ListBoxSelected.Items.Add(inicls.GetIniValue("Display Format", "ListBoxSelected.Items" + Convert.ToString(i), path));
                    }
                    Button6.BackColor = Color.FromArgb(Convert.ToInt32(inicls.GetIniValue("Alarm Type", "Button6.BackColor", path)));
                    Button7.BackColor = Color.FromArgb(Convert.ToInt32(inicls.GetIniValue("Alarm Type", "Button7.BackColor", path)));
                    Button8.BackColor = Color.FromArgb(Convert.ToInt32(inicls.GetIniValue("Alarm Type", "Button8.BackColor", path)));
                    Button9.BackColor = Color.FromArgb(Convert.ToInt32(inicls.GetIniValue("Alarm Type", "Button9.BackColor", path)));
                    Button5.BackColor = Color.FromArgb(Convert.ToInt32(inicls.GetIniValue("Alarm Type", "BackGround.BackColor", path)));
                    ChkAlarmOff.Checked = bool.Parse(inicls.GetIniValue("Alarm Status", "ChkAlarmOff", path));
                    ChkAlarmOn.Checked = bool.Parse(inicls.GetIniValue("Alarm Status", "ChkAlarmOn", path));
                    ChkAlarmVariation.Checked = bool.Parse(inicls.GetIniValue("Alarm Status", "ChkAlarmVariation", path));
                    ChkAlarmAck.Checked = bool.Parse(inicls.GetIniValue("Alarm Status", "ChkAlarmAck", path));

                }
                else
                {

                }
            }
            catch (Exception ex)
            {
                return;
            }

        }

        private void Button6_Click(dynamic sender, EventArgs e)
        {
            try
            {


                ColorDialog1.ShowDialog();

                if (sender.Name == "Button6")
                {
                    Button6.BackColor = ColorDialog1.Color;
                    m_ControlToEdit._nListViewColumnsColor.Add(Button6.BackColor);
                    inicls.SetIniValue("Alarm Type", "Button6.BackColor", Button6.BackColor.ToArgb().ToString(), path);

                }

                if (sender.Name == "Button7")
                {
                    Button7.BackColor = ColorDialog1.Color;
                    m_ControlToEdit._nListViewColumnsColor.Add(Button7.BackColor);
                    inicls.SetIniValue("Alarm Type", "Button7.BackColor", Button7.BackColor.ToArgb().ToString(), path);

                }
                if (sender.Name == "Button8")
                {
                    Button8.BackColor = ColorDialog1.Color;
                    m_ControlToEdit._nListViewColumnsColor.Add(Button8.BackColor);
                    inicls.SetIniValue("Alarm Type", "Button8.BackColor", Button8.BackColor.ToArgb().ToString(), path);

                }
                if (sender.Name == "Button9")
                {
                    Button9.BackColor = ColorDialog1.Color;
                    m_ControlToEdit._nListViewColumnsColor.Add(Button9.BackColor);
                    inicls.SetIniValue("Alarm Type", "Button9.BackColor", Button9.BackColor.ToArgb().ToString(), path);

                }
                if (sender.Name == "Button5")
                {
                    Button5.BackColor = ColorDialog1.Color;
                    m_ControlToEdit._nListViewColumnsColor.Add(Button5.BackColor);
                    inicls.SetIniValue("Alarm Type", "BackGround.BackColor", Button5.BackColor.ToArgb().ToString(), path);

                }
            }
            catch (Exception ex)
            {
                return;
            }
        }

        #endregion
        private List<TabPage> colRemovedTabs = new List<TabPage>();
        public void HidePage(string NamePage)
        {
            // Hide multiple pages
            foreach (TabPage tabpg in TabControl1.TabPages)
            {
                if (tabpg.Name == NamePage)
                {
                    colRemovedTabs.Add(tabpg);
                    TabControl1.Controls.Remove(tabpg);
                }
            }
        }
        public void ShowAllPages()
        {
            //  Reinsert all removed pages
            foreach (TabPage savedTab in colRemovedTabs)
            {
                TabControl1.Controls.Add(savedTab);
            }

            // Clear the collection as it has now done its work
            colRemovedTabs.Clear();

        }

        private string SelectTab;
        private void BtnCheckAll_Click(object sender, EventArgs e)
        {

            try
            {


                switch (SelectTab)
                {
                    case "tbAlarmLevel":

                        CheckBox_Check(tbAlarmLevel);
                        break;
                    case "tbAlarmZone":
                        CheckBox_Check(tbAlarmZone);
                        break;
                    case "tbDisplayType":
                        CheckBox_Check(tbDisplayType);
                        break;
                    case "tbAlarmStatus":
                        CheckBox_Check(tbAlarmStatus);
                        break;
                }
            }
            catch (Exception ex)
            {
                return;
            }


        }
        public void CheckBox_Check(Control root)
        {
            foreach (Control ctrl in root.Controls)
            {
                CheckBox_Check(ctrl);
                if (ctrl is CheckBox)
                {
                    ((CheckBox)ctrl).Checked = true;
                }
            }
        }
        public void CheckBox_UCheck(Control root)
        {
            foreach (Control ctrl in root.Controls)
            {
                CheckBox_UCheck(ctrl);
                if (ctrl is CheckBox)
                {
                    ((CheckBox)ctrl).Checked = false;
                }
            }
        }
        private void CheckBox17_CheckedChanged(object sender, EventArgs e)
        {

        }

        private void BtnUCheckAll_Click(object sender, EventArgs e)
        {
            try
            {
                switch (SelectTab)
                {
                    case "tbAlarmLevel":
                        CheckBox_UCheck(tbAlarmLevel);
                        break;
                    case "tbAlarmZone":
                        CheckBox_UCheck(tbAlarmZone);
                        break;
                    case "tbDisplayType":
                        CheckBox_UCheck(tbDisplayType);
                        break;
                    case "tbAlarmStatus":
                        CheckBox_UCheck(tbAlarmStatus);
                        break;
                }
            }
            catch (Exception ex)
            {
                return;
            }



        }

        private void ListBoxSelected_DrawItem(dynamic sender, DrawItemEventArgs e)
        {
            try
            {



                //' الخروج من الإجراء إذا لم يوجد عناصر
                if (e.Index < 0)
                {
                    return;
                }

                //' اللون الإفتراضي لخلفية العنصر المحدد
                e.DrawBackground();

                //' خصائص تنسيق النص
                StringFormat _sf = new StringFormat();

                //' تحويل الإتجاه من اليمين إلى اليسار
                if (sender.RightToLeft)
                {
                    _sf.FormatFlags = StringFormatFlags.DirectionRightToLeft;
                }


                //' للعنصر المحدد
                if ((e.State & DrawItemState.Selected) == DrawItemState.Selected)
                {

                    //' لون خلفية العنصر المحدد
                    e.Graphics.FillRectangle(Brushes.Brown, e.Bounds);

                    //' لون نص العنصر المحدد
                    e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, new SolidBrush(e.ForeColor), e.Bounds, _sf);


                }
                else //' للعناصر غير المحددة
                {

                    if ((e.Index % 2) != 0) // الترتيب الزوجي -------------------------
                    {

                        //' لون خلفية العنصر  غير المحدد حسب الترتيب الزوجي
                        e.Graphics.FillRectangle(new SolidBrush(Color.FromArgb(10, 0, 0, 0)), e.Bounds);

                        //' لون نص العنصر بالترتيب الزوجي
                        e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, Brushes.Red, e.Bounds, _sf);

                    }
                    else // الترتيب الفردي -------------------------
                    {

                        //' لون خلفية العنصر  غير المحدد حسب الترتيب الفردي
                        e.Graphics.FillRectangle(new SolidBrush(Color.FromArgb(20, 0, 0, 0)), e.Bounds);

                        //' لون نص العنصر بالترتيب الفردي
                        e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, Brushes.Blue, e.Bounds, _sf);
                    }

                }

                //' رسم مستطيل التركيز
                e.DrawFocusRectangle();

            }
            catch (Exception ex)
            {
                return;
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            try
            {

                string Selected_txt = Convert.ToString(ListBoxNonSelected.SelectedItem);

                if (Selected_txt != null) // التحقق من عدم خلو اختيار قيمة فارغة
                {

                    if (!this.ListBoxSelected.Items.Contains(Selected_txt)) // يتم التحقق هل النص موجود سلفا باللست بوكس أما لا حتى لا يضاف مرة أخرى
                    {

                        this.ListBoxSelected.Items.Add(Selected_txt);
                        ListBoxNonSelected.Items.RemoveAt(ListBoxNonSelected.SelectedIndex);
                    }

                }
            }
            catch (Exception ex)
            {
                return;
            }


        }


        private void tbAlarmLevel_Enter(dynamic sender, EventArgs e)
        {
            try
            {
                if (sender.name == "tbAlarmLevel")
                {
                    SelectTab = tbAlarmLevel.Name;
                }
                if (sender.name == "tbAlarmZone")
                {
                    SelectTab = tbAlarmZone.Name;
                }
                if (sender.name == "tbDisplayType")
                {
                    SelectTab = tbDisplayType.Name;
                }
                if (sender.name == "tbAlarmStatus")
                {
                    SelectTab = tbAlarmStatus.Name;
                }

            }
            catch (Exception ex)
            {
                return;
            }

        }
    }
}

File: HslScada.Controls_Net45/AlarmMan/Designers/ListViewDesignerForm.vb
Imports System.Deployment
Imports System.Drawing
Imports System.IO
Imports System.Windows.Forms
Imports System.Xml

Public Class ListViewDesignerForm
#Region "Properties"
    Public Property ControlToEdit As AlarmMan
    Public NListViewColumns() As String
    Private SetListViewColumns As String
    Private Sub BtnOK_Click(sender As Object, e As EventArgs) Handles BtnOK.Click
        Try


            Dim ListBoxSelectedItemsCount As String = ListBoxSelected.Items.Count
            For i As Integer = 0 To ListBoxSelected.Items.Count - 1
                SetListViewColumns = String.Format("{0}.", ListBoxSelected.Items(i))
            Next
            My.Settings.ChkAlarmOff = ChkAlarmOff.Checked
            My.Settings.ChkAlarmOn = ChkAlarmOn.Checked
            My.Settings.ChkAlarmVariation = ChkAlarmVariation.Checked
            My.Settings.ChkAlarmAck = ChkAlarmAck.Checked


            My.Settings.Save()

            Close()
        Catch ex As Exception
            Exit Sub
        End Try
    End Sub

    Private Sub BtnCancel_Click(sender As Object, e As EventArgs) Handles BtnCancel.Click
        Close()
    End Sub
    Dim Lst As New ListBox

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Try
            Dim Selected_txt As String = ListBoxSelected.SelectedItem

            If Selected_txt <> vbNullString Then ' التحقق من عدم خلو اختيار قيمة فارغة 

                If Not Me.ListBoxNonSelected.Items.Contains(Selected_txt) Then ' يتم التحقق هل النص موجود سلفا باللست بوكس أما لا حتى لا يضاف مرة أخرى 

                    Me.ListBoxNonSelected.Items.Add(Selected_txt)
                    ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex)
                End If

            End If

        Catch ex As Exception
            Exit Sub
        End Try


    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Try

            Dim MyIndex As Integer
            Dim MyItem As String
            MyIndex = ListBoxSelected.SelectedIndex
            MyItem = ListBoxSelected.Text

            If MyIndex = 0 Then
                Exit Sub
            End If

            ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex)
            ListBoxSelected.Items.Insert(MyIndex - 1, MyItem)
            ListBoxSelected.SelectedIndex = MyIndex - 1
        Catch ex As Exception
            Exit Sub
        End Try



    End Sub

    Private Sub ListBoxSelected_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ListBoxSelected.SelectedIndexChanged
        'If (ListBoxNonSelected.Items.Contains(ListBoxSelected.SelectedItem)) Then
        '    '' The item Is already exist in the listbox collection 
        '    Exit Sub
        'Else
        '    '' The item Is Not exist in the listbox collection 
        '    ListBoxNonSelected.Items.Add(ListBoxSelected.SelectedItem)
        'End If
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Try
            Dim MyIndex As Integer
            Dim MyItem As String
            MyIndex = ListBoxSelected.SelectedIndex
            MyItem = ListBoxSelected.Text

            If MyIndex = ListBoxSelected.Items.Count - 1 Then
                Exit Sub
            End If

            ListBoxSelected.Items.RemoveAt(ListBoxSelected.SelectedIndex)
            ListBoxSelected.Items.Insert(MyIndex + 1, MyItem)
            ListBoxSelected.SelectedIndex = MyIndex + 1
        Catch ex As Exception
            Exit Sub
        End Try

    End Sub

    Dim nListBoxSelected As New List(Of String)
    Private Sub ListViewDesignerForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Try


            For i As Integer = 0 To ListBoxSelected.Items.Count - 1
                nListBoxSelected.Add(ListBoxSelected.Items(i))
            Next
            ControlToEdit.NListViewColumnsColor.Clear()
            ListBoxSelected.Items.Clear()
            ' Open the file to read from.


            NListViewColumns = My.Settings.ListViewColumns.Split(New [Char]() {"."c})

            For Each item As String In NListViewColumns
                ListBoxSelected.Items.Add(item)

            Next



            Button6.BackColor = My.Settings.Button6BackColor
                Button7.BackColor = My.Settings.Button7BackColor
                Button8.BackColor = My.Settings.Button8BackColor
                Button9.BackColor = My.Settings.Button9BackColor
                Button5.BackColor = My.Settings.BackGroundBackColor
                ChkAlarmOff.Checked = My.Settings.ChkAlarmOff
                ChkAlarmOn.Checked = My.Settings.ChkAlarmOn
                ChkAlarmVariation.Checked = My.Settings.ChkAlarmVariation
                ChkAlarmAck.Checked = My.Settings.ChkAlarmAck


        Catch ex As Exception
            Exit Sub
        End Try

    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button9.Click, Button8.Click, Button7.Click, Button6.Click, Button5.Click
        Try


            ColorDialog1.ShowDialog()

            If sender.Name = "Button6" Then
                Button6.BackColor = ColorDialog1.Color
                ControlToEdit.NListViewColumnsColor.Add(Button6.BackColor)
                My.Settings.Button6BackColor = Button6.BackColor


            End If

            If sender.Name = "Button7" Then
                Button7.BackColor = ColorDialog1.Color
                ControlToEdit.NListViewColumnsColor.Add(Button7.BackColor)
                My.Settings.Button7BackColor = Button7.BackColor

            End If
            If sender.Name = "Button8" Then
                Button8.BackColor = ColorDialog1.Color
                ControlToEdit.NListViewColumnsColor.Add(Button8.BackColor)
                My.Settings.Button8BackColor = Button8.BackColor

            End If
            If sender.Name = "Button9" Then
                Button9.BackColor = ColorDialog1.Color
                ControlToEdit.NListViewColumnsColor.Add(Button9.BackColor)
                My.Settings.Button9BackColor = Button9.BackColor

            End If
            If sender.Name = "Button5" Then
                Button5.BackColor = ColorDialog1.Color
                ControlToEdit.NListViewColumnsColor.Add(Button5.BackColor)
                My.Settings.BackGroundBackColor = Button5.BackColor

            End If
        Catch ex As Exception
            Exit Sub
        End Try
        My.Settings.Save()
    End Sub

#End Region
    Dim colRemovedTabs As New List(Of TabPage)
    Sub HidePage(ByVal NamePage As String)
        ' Hide multiple pages
        For Each tabpg As TabPage In TabControl1.TabPages
            If tabpg.Name = NamePage Then
                colRemovedTabs.Add(tabpg)
                TabControl1.Controls.Remove(tabpg)
            End If
        Next
    End Sub
    Sub ShowAllPages()
        '  Reinsert all removed pages
        For Each savedTab As TabPage In colRemovedTabs
            TabControl1.Controls.Add(savedTab)
        Next

        ' Clear the collection as it has now done its work
        colRemovedTabs.Clear()

    End Sub

    Dim SelectTab As String
    Private Sub BtnCheckAll_Click(sender As Object, e As EventArgs) Handles BtnCheckAll.Click

        Try


            Select Case SelectTab
                Case "tbAlarmLevel"

                    CheckBox_Check(tbAlarmLevel)
                Case "tbAlarmZone"
                    CheckBox_Check(tbAlarmZone)
                Case "tbDisplayType"
                    CheckBox_Check(tbDisplayType)
                Case "tbAlarmStatus"
                    CheckBox_Check(tbAlarmStatus)
            End Select
        Catch ex As Exception
            Exit Sub
        End Try


    End Sub
    Public Sub CheckBox_Check(ByVal root As Control)
        For Each ctrl As Control In root.Controls
            CheckBox_Check(ctrl)
            If TypeOf ctrl Is CheckBox Then
                CType(ctrl, CheckBox).Checked = True
            End If
        Next
    End Sub
    Public Sub CheckBox_UCheck(ByVal root As Control)
        For Each ctrl As Control In root.Controls
            CheckBox_UCheck(ctrl)
            If TypeOf ctrl Is CheckBox Then
                CType(ctrl, CheckBox).Checked = False
            End If
        Next
    End Sub
    Private Sub CheckBox17_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox17.CheckedChanged

    End Sub

    Private Sub BtnUCheckAll_Click(sender As Object, e As EventArgs) Handles BtnUCheckAll.Click
        Try
            Select Case SelectTab
                Case "tbAlarmLevel"
                    CheckBox_UCheck(tbAlarmLevel)
                Case "tbAlarmZone"
                    CheckBox_UCheck(tbAlarmZone)
                Case "tbDisplayType"
                    CheckBox_UCheck(tbDisplayType)
                Case "tbAlarmStatus"
                    CheckBox_UCheck(tbAlarmStatus)
            End Select
        Catch ex As Exception
            Exit Sub
        End Try



    End Sub

    Private Sub ListBoxSelected_DrawItem(sender As Object, e As DrawItemEventArgs) Handles ListBoxSelected.DrawItem
        Try



            '' الخروج من الإجراء إذا لم يوجد عناصر
            If e.Index < 0 Then Exit Sub

            '' اللون الإفتراضي لخلفية العنصر المحدد
            e.DrawBackground()

            '' خصائص تنسيق النص
            Dim _sf As New StringFormat

            '' تحويل الإتجاه من اليمين إلى اليسار
            If sender.RightToLeft Then _sf.FormatFlags = StringFormatFlags.DirectionRightToLeft


            '' للعنصر المحدد
            If (e.State And DrawItemState.Selected) = DrawItemState.Selected Then

                '' لون خلفية العنصر المحدد
                e.Graphics.FillRectangle(Brushes.Brown, e.Bounds)

                '' لون نص العنصر المحدد
                e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, New SolidBrush(e.ForeColor), e.Bounds, _sf)


            Else '' للعناصر غير المحددة

                If e.Index Mod 2 Then ' الترتيب الزوجي -------------------------

                    '' لون خلفية العنصر  غير المحدد حسب الترتيب الزوجي
                    e.Graphics.FillRectangle(New SolidBrush(Color.FromArgb(10, 0, 0, 0)), e.Bounds)

                    '' لون نص العنصر بالترتيب الزوجي
                    e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, Brushes.Red, e.Bounds, _sf)

                Else ' الترتيب الفردي -------------------------

                    '' لون خلفية العنصر  غير المحدد حسب الترتيب الفردي
                    e.Graphics.FillRectangle(New SolidBrush(Color.FromArgb(20, 0, 0, 0)), e.Bounds)

                    '' لون نص العنصر بالترتيب الفردي
                    e.Graphics.DrawString(sender.GetItemText(sender.Items(e.Index)), e.Font, Brushes.Blue, e.Bounds, _sf)
                End If

            End If

            '' رسم مستطيل التركيز
            e.DrawFocusRectangle()

        Catch ex As Exception
            Exit Sub
        End Try
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Try

            Dim Selected_txt As String = ListBoxNonSelected.SelectedItem

            If Selected_txt <> vbNullString Then ' التحقق من عدم خلو اختيار قيمة فارغة 

                If Not Me.ListBoxSelected.Items.Contains(Selected_txt) Then ' يتم التحقق هل النص موجود سلفا باللست بوكس أما لا حتى لا يضاف مرة أخرى 

                    Me.ListBoxSelected.Items.Add(Selected_txt)
                    ListBoxNonSelected.Items.RemoveAt(ListBoxNonSelected.SelectedIndex)
                End If

            End If
        Catch ex As Exception
            Exit Sub
        End Try


    End Sub


    Private Sub tbAlarmLevel_Enter(sender As Object, e As EventArgs) Handles tbAlarmLevel.Enter, tbAlarmZone.Enter, tbDisplayType.Enter, tbAlarmStatus.Enter
        Try
            If sender.name = "tbAlarmLevel" Then SelectTab = tbAlarmLevel.Name
            If sender.name = "tbAlarmZone" Then SelectTab = tbAlarmZone.Name
            If sender.name = "tbDisplayType" Then SelectTab = tbDisplayType.Name
            If sender.name = "tbAlarmStatus" Then SelectTab = tbAlarmStatus.Name

        Catch ex As Exception
            Exit Sub
        End Try

    End Sub
End Class